libs/pruviloj/Pruviloj/Core.idr

```diff
@@ -268,3 +274,37 @@ reflexivity =
                 , NamePart `{reflexivity}
                 , TextPart "is not applicable."
                 ]
+
+||| Attempt to swap the sides of equality in a goal.
+symmetry : Elab ()
+symmetry = do
+    [_,_,_,_,_] <- apply (Var `{{sym}}) [True, True, True, True, False]
+       | _ => fail [TextPart "Failed while applying", NamePart `{symmetry} ]
+    solve
+    attack
+
+||| Swap the sides of an equality and bind the result to supplied name.
+symmetryIn : Raw -> TTName -> Elab ()
+symmetryIn t n = do
+    tt <- getTTType t
+    case tt of
+      `((=) {A=~A} {B=~B} ~l ~r) => do
+        af <- forget A
+        bf <- forget B
+        lf <- forget l
+        rf <- forget r
+        let ts = the Raw $ `(sym {a=~af} {b=~bf} {left=~lf} {right=~rf} ~t)
+        tsty <- forget !(getTTType ts)
+        letbind n tsty ts
+      _ => fail [TermPart tt, TextPart "is not an equality"]
```

Comment:
Instead of `_` on the left hand side, give it a name, and then put another `TermPart` in the message.